src/test/util/mod.rs

```diff
@@ -267,19 +270,19 @@ impl TestClient {
     }
 
     pub fn is_standalone(&self) -> bool {
-        !self.is_replica_set() && !self.is_sharded()
+        self.base_topology() == Topology::Single
     }
 
     pub fn is_replica_set(&self) -> bool {
-        self.options.repl_set_name.is_some()
+        self.base_topology() == Topology::ReplicaSet
     }
 
     pub fn is_sharded(&self) -> bool {
-        self.server_info.msg.as_deref() == Some("isdbgrid")
```

Comment:
This returns `true` for load-balanced toplogies, so to avoid having to manually maintain a bunch of fiddly boolean exclusions I refactored these.